main.py

```diff
@@ -644,10 +646,14 @@ async def scrape_channel_media(
                 )
             )
 
+    download_semaphore = asyncio.Semaphore(value=MAXIMUM_CONCURRENT_DOWNLOADS)
+
     # Save all files if not in cache
     async def dl_file(attachment: Attachment, attachment_filepath: str) -> None:
+        await download_semaphore.acquire()
         if not os.path.exists(attachment_filepath):
             await attachment.save(attachment_filepath)
+        download_semaphore.release()
```

Comment:
We may want to use the `async with` syntax detailed [in the sephamore docs](https://docs.python.org/3/library/asyncio-sync.html#asyncio.Semaphore), as it guarantees that the semaphore is released even if an Exception is raised?

Also we probably don't need to take a slot if the file does end up existing.

```suggestion
        if os.path.exists(attachment_filepath):
            return
        
        # Limit concurrent downloads
        await with download_semaphore:
            await attachment.save(attachment_filepath)
```